src/sql/src/pure.rs

```diff
@@ -1523,39 +1546,113 @@ async fn purify_create_table_from_source(
             purified_export
         }
         GenericSourceConnection::LoadGenerator(load_gen_connection) => {
-            let mut views = load_gen_connection.load_generator.views();
-            // Resolve the desired view reference to a `schema_name.view_name` reference
-            let resolver = SourceReferenceResolver::new(
-                LOAD_GENERATOR_DATABASE_NAME,
-                &views
-                    .iter()
-                    .map(|(view_name, _, _)| {
-                        (load_gen_connection.load_generator.schema_name(), *view_name)
-                    })
-                    .collect_vec(),
-            )?;
-            let (qualified_reference, view_idx) = resolver.resolve(&external_reference.0, 2)?;
+            // TODO(roshan): Refactor KeyValue load generator rendering to support multiple outputs
+            if matches!(
+                load_gen_connection.load_generator,
+                mz_storage_types::sources::load_generator::LoadGenerator::KeyValue(_)
+            ) {
+                sql_bail!(
+                    "{} is a {} source, which does not support CREATE TABLE .. FROM SOURCE.",
+                    scx.catalog.minimal_qualification(qualified_source_name),
+                    connection_name
+                )
+            };
 
-            let (_, desired_view_desc, output) = views.swap_remove(view_idx);
+            let mut views = load_gen_connection.load_generator.views();
+            if views.is_empty() {
```

Comment:
Continuing with my comment above, we should get each source to present its available references and have a common piece of source-agnostic code that selects a subset of them based on what the user typed.